### DISCUSSION TITLE
How to add SSL Cert to nginx-s3-gateway

### DISCUSSION THREAD

**Discussion by tcarolan** [nginxinc/nginx-s3-gateway#138] [Q&A] [answered]
I know the documentation says the gateway does not support SSL/TLS, but I still would like to add it. I am using docker and have everything working over regular HTTP. How can I add a security certificate?

**Comment by dekobon** [comment]
Right now, you will have to extend the container image by creating a new Dockerfile and using `FROM`. In the new container, you will create a file that replaces the file `/etc/nginx/templates/gateway/s3_server.conf.template`. Then in that file you can add the typical TLS parameters like:
```
    listen      443 rcvbuf=8192 backlog=4096 ssl http2 default_server;
    listen [::]:443 rcvbuf=8192 backlog=4096 ssl http2 default_server;
    server_name {TLS_HOSTNAME};

    # Reject handshakes for server names other than {TLS_HOSTNAME}
    # TODO: Figure out how to enable this
    # ssl_reject_handshake on;

    # Let's Encrypt configuration
    ssl_certificate         /etc/letsencrypt/live/{TLS_HOSTNAME}/fullchain.pem;
    ssl_certificate_key     /etc/letsencrypt/live/{TLS_HOSTNAME}/privkey.pem;
    ssl_trusted_certificate /etc/letsencrypt/live/{TLS_HOSTNAME}/chain.pem;

    include /etc/nginx/options-ssl-nginx.conf;

    # OCSP stapling
    ssl_stapling on;
    ssl_stapling_verify on;
```

Please let us know how it goes.

**Reply by tcarolan** [reply, marked as answer]
Thank you! I actually got this to work.
